### PR TITLE
Get node's IPv6 address with non-calico CNI

### PR DESCRIPTION
**Pull request by tomastigera**
Without calico cni, Node.Spec.BGP is not populated. In v4 mode, libcalico falls back to getting the node's address from Node.Spec.Addresses. IPv6 detection takes a different path and always depended on Node.Spec.BPG being present. Do the same fallback in DataplanePassthru where the v6 updates are created.

This guarantees that we always know the node's IP in k8s environment. eBPF mode needs to know that to load its hep programs.

refs https://github.com/projectcalico/calico/issues/4736

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
Node learns about it's ipv6 address in kubernetes even if BGP is turned off and CNI is not calico.
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: felix/calc/dataplane_passthru.go
Comment (fasaxc) [inline felix/calc/dataplane_passthru.go:182]: Same here; could check if IP is in h.hostIPv6

**Review by fasaxc — APPROVED** [review]
Looks like it should do the trick but I wonder if we should move it to its own calc graph node.  The logic is starting to get a bit complex for the "pass through", which was mean to be a simple stateless component for passing through trivial objects.

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Comment by tomastigera** [comment]
only node-ci failed

**Pull request merged**
